File: ai/src/main/java/org/pingel/causality/examples/Model3dot8b.java
/*
 * Created on Jun 7, 2005
 *
 */
package org.pingel.causality.examples;

import org.pingel.bayes.CausalModel;
import org.pingel.bayes.Function;
import org.pingel.bayes.ModelVisualizer;
import org.pingel.bayes.RandomVariable;

public class Model3dot8b extends CausalModel {

    public Model3dot8b()
    {
        super("3.8b");

        RandomVariable X = addVariable(new RandomVariable("X", "x"));
        RandomVariable Y = addVariable(new RandomVariable("Y", "y"));
        RandomVariable Z = addVariable(new RandomVariable("Z", "z"));
        
        RandomVariable U = addVariable(new RandomVariable("U", "u", false));

        addFunction(new Function(Y, X, Z, U));
        addFunction(new Function(Z, X, U));
    }
    
    public static void main(String[] argv)
    {
        CausalModel model = new Model3dot8b();
        ModelVisualizer.draw(model);
    }
    

}

File: ai/src/main/scala/org/pingel/causality/examples/Model3dot8b.scala
/*
 * Created on Jun 7, 2005
 *
 */
package org.pingel.causality.examples

import org.pingel.causality.CausalModel
import org.pingel.forms.Basic.PFunction
import org.pingel.bayes.RandomVariable

object Model3dot8b extends CausalModel("3.8b") {

  val X = new RandomVariable("X")
  val Y = new RandomVariable("Y")
  val Z = new RandomVariable("Z")
  val U = new RandomVariable("U", None, false)

  g ++= (X :: Y :: Z :: U :: Nil)

  addFunction(new PFunction(Y, List(X, Z, U)))
  addFunction(new PFunction(Z, List(X, U)))

  def main(args: Array[String]) {
    g.draw
  }
}
